library/init/algebra/order.lean

```diff
@@ -187,31 +187,13 @@ section linear_order
 ### Definition of `linear_order` and lemmas about types with a linear order
 -/
 
-/-- Default definition of `max`. -/
-def max_default {α : Type u} [has_le α] [decidable_rel ((≤) : α → α → Prop)] (a b : α) :=
-if b ≤ a then a else b
-
-/-- Default definition of `min`. -/
-def min_default {α : Type u} [has_le α] [decidable_rel ((≤) : α → α → Prop)] (a b : α) :=
-if a ≤ b then a else b
-
 /-- A linear order is reflexive, transitive, antisymmetric and total relation `≤`.
 We assume that every linear ordered type has decidable `(≤)`, `(<)`, and `(=)`. -/
 class linear_order (α : Type u) extends partial_order α :=
 (le_total : ∀ a b : α, a ≤ b ∨ b ≤ a)
-(decidable_le : decidable_rel (≤))
-(decidable_eq : decidable_eq α := @decidable_eq_of_decidable_le _ _ decidable_le)
-(decidable_lt : decidable_rel ((<) : α → α → Prop) :=
-    @decidable_lt_of_decidable_le _ _ decidable_le)
-(max : α → α → α := @max_default α _ _)
```

Comment:
I think we should still keep `max` in `linear_order` (or maybe add a `has_sup` class) so that you can override it with a more efficient (or defeq) version.